### ISSUE TITLE
knexfile.js for migrations

### ISSUE THREAD

**Issue by escobar5** [feathersjs/feathers-knex#116]
I'm trying to configure knex migrations in my feathersjs app, I need to create a knexfile with the configurations for knex migrations to work, but I don't want to have my database configuration in two places (feathersjs config and knexfile), does anybody know how handle database configuration for both?

**Comment by daffl** [comment]
You can `const app = require('./src/app`);` in the Knexfile and then get either the Knex client itself via `app.get('knexClient')` or the database configuration via `app.get('<db-name>')`.

**Comment by escobar5** [comment]
You're right, that should work. Thanks

**Comment by kodayashi** [comment]
This helped me too, but @daffl I'm trying to sort out what the best practice is here. Migrating and seeding is a key feature, so would it make sense for feathers-knex to do this type of setup when installed?

In addition, the knex CLI isn't available when installed via the instructions in the README, is the preferred approach to also install knex globally to leverage the migration/seed apis?

Thanks!

**Comment by daffl** [comment]
Feathers adapters try to be as little in the way of the normal usage of the ORM/database as possible. So whatever the Knex documentation recommends will also work here. The main concern of this adapter is to connect a Knex model to the service interface and implement the common querying mechanism but that is pretty much it.

**Comment by kodayashi** [comment]
Cool, fair enough. Didn't see anything in the knex docs to accommodate the feathers setup but like I said the above recommendation worked great. To get migrations/seeding working i just installed knex globally, ran `knex init` and modified the `knexfile.js` file to:

```javascript
const app = require('./src/app');

module.exports = app.get('postgres');
```

Then I could use the standard default/production feathers config files and add migration instructions there, e.g.:

```javascript
  "postgres": {
    "client": "pg",
    "connection": "postgres://user:password@localhost:5432/database",
    "seeds": {
      "directory": "../src/seeds"
    }
```

I believe migrations will work as well with this strategy.

Thanks for the reply man!

**Comment by daffl** [comment]
If you would like to add that information to the Readme, a pull request would be very welcome :)

**Comment by carquiza** [comment]
Thanks for the solution. May I add that it seems to be necessary to also comment out the parts in the models that create tables, as they seem to run before the migrations are called.

**Comment by soulprovidr** [comment]
I know this is an old issue, but I echo the sentiment above re: migrations being a fundamental part of an application. This should really be addressed by the tool and in the documentation.

**Comment by daffl** [comment]
It's about adding the information above which can be done by [editing the readme](https://github.com/feathersjs-ecosystem/feathers-knex/edit/master/README.md).

**Comment by jrj2211** [comment]
I'd like to make a suggestion - I'm using typescript for my project and the provided method in the readme did not work well. I kept running into typescript errors when trying to require ./src/app. I installed typescript-require but  that didn't really seem to work but I found a simpler way:

```
const config = require('config');
module.exports = config.get('postgres');
```

Can we update the readme to this as it supports both normal js and typescript without having to install any additional packages. It also is more efficient as theres in most cases probably no reason to load the entire app just for the knexfile. Correct me if I'm wrong though, this is my first featherjs app.
